src/Ionide.ProjInfo/InspectSln.fs

```diff
@@ -112,23 +128,45 @@ module InspectSln =
 
             // three kinds of items - projects, folders, items
             // yield them all here
-            let projectsWeCareAbout =
-                match projectsToRead with
-                | None -> sln.SolutionProjects :> seq<_>
-                | Some filteredProjects ->
-                    sln.SolutionProjects
-                    |> Seq.filter (fun slnProject -> filteredProjects.Contains(makeAbsoluteFromSlnDir slnProject.FilePath))
-
             let allItems = [
+                // Projects at solution level get returned directly
                 yield!
                     projectsWeCareAbout
+                    |> Seq.filter (
+                        _.Parent
+                        >> isNull
+                    )
                     |> Seq.map parseProject
+
+                // parseFolder will parse any projects or folders within the specified folder itself, so just process the root folders here
                 yield!
                     sln.SolutionFolders
+                    |> Seq.filter (
+                        _.Parent
+                        >> isNull
+                    )
                     |> Seq.map parseFolder
+
+                // 'SolutionItems' contains all of SolutionFolders and SolutionProjects, so only include things that aren't in those to avoid duplication
```

Comment:
When testing I noticed that some solution items are being returned twice here, which didn't seem right (note the duplicated guids in the results, with different Kinds)

<img width="1318" height="226" alt="image" src="https://github.com/user-attachments/assets/d44ba2da-7b98-4f75-9277-25df36ce60e1" />

Looks to be because 'SolutionItems' contains all the folders and so the existing logic will include any root level folders as 'Unknown' as well as including them as folders